crates/holochain_state/src/mutations.rs

```diff
@@ -601,3 +604,140 @@ pub fn unlock_chain(txn: &mut Transaction) -> StateMutationResult<()> {
     txn.execute("DELETE FROM ChainLock", [])?;
     Ok(())
 }
+
+pub fn delete_all_ephemeral_scheduled_fns(txn: &mut Transaction) -> StateMutationResult<()> {
+    txn.execute("DELETE FROM ScheduledFunctions WHERE ephemeral = true", [])?;
+    Ok(())
+}
+
+pub fn delete_live_ephemeral_scheduled_fns(
+    txn: &mut Transaction,
+    now: Timestamp,
+) -> StateMutationResult<()> {
+    txn.execute(
+        "DELETE FROM ScheduledFunctions WHERE ephemeral = true AND start <= ?",
+        [now],
+    )?;
+    Ok(())
+}
+
+pub fn reschedule_expired(txn: &mut Transaction, now: Timestamp) -> StateMutationResult<()> {
+    let rows = {
+        let mut stmt = txn.prepare(
+            "
+            SELECT
+            zome_name,
+            scheduled_fn,
+            maybe_schedule
+            FROM ScheduledFunctions
+            WHERE
+            NOT ephemeral
+            AND end < ?",
```

Comment:
consider pulling your sql out into separate `.sql` files https://github.com/holochain/holochain/tree/develop/crates/holochain_sqlite/src/sql/cell so it can be formatted / linted. You can expose it publicly to this crate from here: https://github.com/holochain/holochain/blob/develop/crates/holochain_sqlite/src/sql.rs